core/src/com/unciv/logic/map/mapunit/movement/UnitMovement.kt

```diff
@@ -238,8 +238,14 @@ class UnitMovement(val unit: MapUnit) {
      * @return The tile that we reached this turn
      */
     fun headTowards(destination: Tile): Tile {
+        val escortUnit = if (unit.getOtherEscortUnit() != null && unit.escorting)
+            unit.getOtherEscortUnit() else null
```

Comment:
```suggestion
        val escortUnit = if (unit.escorting) unit.getOtherEscortUnit() else null
```
since if getOtherEscortUnit is null, escortUnit will be null anyway, no need to have a separate check